### PR TITLE
Parses a CSV image annotation report file and adjusts all annotation coordinates to be inside the image 

### PR DESCRIPTION
**Pull request by lombardata**

### CODE REVIEW

**Comment by lombardata** [comment]
Parses a CSV image annotation report file and adjusts all annotation coordinates to be inside the image

File: README.md [hunk 1]
```diff
@@ -1,26 +1,13 @@
-# BIIGLE Community Scripts
```
Comment (mzur) [inline README.md:1]: You accidentally replaced the global readme of this repository. Please create a subdirectory (with your script and readme) and update the list in the global readme.

File: check_coordinates.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import json
```
Comment (mzur) [inline check_coordinates.py:1]: Please move this script to the new subdirectory, too (see above).

File: check_coordinates.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            elif x_coord[i] > max_width :
+                x_coord[i] = max_width
+        # check that y coordinates are in the image range
+        for i in range(len(y_coord)) :
+            if y_coord[i] < 0 :
+                y_coord[i] = 0.0
+            elif y_coord[i] > max_heigth :
+                y_coord[i] = max_heigth
+        # create the adjusted coordinates vector
+        desired_array_adjusted = x_coord
```
Comment (mzur) [inline check_coordinates.py:52]: Possible alternative implementation:

```suggestion
        x_coord = map(lambda x: max(min(x, max_width), 0), desired_array[::2])
        y_coord = map(lambda y: max(min(y, max_heigth), 0), desired_array[1::2])
        # create the adjusted coordinates vector
        desired_array_adjusted = list(x_coord)                                                                               
```

File: check_coordinates.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        max_heigth =  desired_dict["height"]
+        max_width =  desired_dict["width"]
+        # get vector of coordinates
+        # row.points is a string, we want to transform it in an array of doubles
+        string_array = row.points
+        string_array = string_array.replace('[', '')
+        string_array = string_array.replace(']', '')
+        splitted_string = string_array.split (",")
+        # desired_array is an array of doubles, like [x1, y1, x2, y2, ecc]
+        desired_array = [float(numeric_string) for numeric_string in splitted_string]
```
Comment (mzur) [inline check_coordinates.py:36]: This should include a check for cirlces (lists with exactly three elements). The third element of the list (radius of the circle) should not be modified by the script.

**Review by mzur — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please see my comments.

**Comment by lombardata** [comment]
thanks for the suggestions!

File: check_coordinates.py (hunk 4 above)
Comment (lombardata) [reply check_coordinates.py:36]: OK, I had not included the circles as I only work with polygons. I added an if condition for circles and an else for all other shapes. I hope this is sufficient

**Pull request closed without merging**